packages/x-data-grid/src/hooks/core/pipeProcessing/gridPipeProcessingApi.ts

```diff
@@ -26,6 +26,7 @@ import { HeightEntry } from '../../features/rows/gridRowsMetaInterfaces';
 export type GridPipeProcessorGroup = keyof GridPipeProcessingLookup;
 
 export interface GridPipeProcessingLookup {
+  clearDataSourceState: { value: null };
```

Comment:
Tbh I don't love pipe processors, I feel like they make the flow a bit hard to understand. Is there a way we could use an alternative? Could the newer grid config context be used instead?

Reply:
Agreed, especially for this specific case, a prop would be more suited.